### PR TITLE
Search

### PR DESCRIPTION
**Pull request by krgarima**
## Change Logs:
# Description: Added Search feature, toast

# More details:

1. Search -
- User can search any product by clicking on `search` icon provided in the Navbar. 

2. Toast - 
- Added toast in `ProductCard` and `WishlistCard`. User is notified when he adds, removes items etc.

3. Authentication with tokens- 
- Earlier, I had implemented authentication without adding tokens to the local storage (checking for correct username and password). Now, the user will only be allowed to access private routes when he has tokens stored in his local.

Screenshot: 
search

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,11 +10,12 @@ import Login from "./Pages/Login/Login";
 import SignUp from "./Pages/SignUp/SignUp";
 import Cart from "./Pages/Cart/Cart";
 import Wishlist from "./Pages/Wishlist/Wishlist";
```
Comment (sakshi006) [inline src/App.js]: import these from a single index.js file

**Review by sakshi006 — APPROVED** [review]
Looks good to merge

**Review by Asuraking1n — APPROVED** [review]
Code looks good, just one thing I suggest instead of using that search button, search products onChange bcz user generally dont know what to search, so when user can see the seacrching the ux experience will be good

File: src/Components/CartCard/CartCard.js [hunk 2]
```diff
@@ -2,11 +2,24 @@ import React, { useContext } from "react";
 import "./CartCard.css";
 import { CartContext } from "../../Context/cart-context.js";
 import { WishlistContext } from "../../Context/wishlist-context";
```
Comment (vikrantkode) [inline src/Components/CartCard/CartCard.js]: import all context from single file

**Review by vikrantkode — APPROVED** [review]
just a small suggestion added .. which you can do for all files Rest is good to merge
Code is clean and easy to read

File: src/Pages/Search/Search.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const { products } = useContext(ProductContext);
+  const [searchList, setSearchList] = useState([]);
+  const [successMsg, setSuccesMsg] = useState("");
+
+  const handleChange = (e) => {
+    e.preventDefault();
+    setSearch(e.target.value);
+  };
+
+  const handleSearch = (e) => {
```
Comment (ivinayakg) [inline src/Pages/Search/Search.js:29]: this function is beautiful, i didn't realize when I was doing this, that we can put the category up there let alone put this whole thing in a different component, amazing thinking

**Review by ivinayakg — APPROVED** [review]
im still amazed by that search functionality, great work.

File: src/Pages/Search/Search.js (hunk 3 above)
Comment (krgarima) [reply src/Pages/Search/Search.js:29]: thank you

File: src/Components/CartCard/CartCard.js (hunk 2 above)
Comment (krgarima) [reply src/Components/CartCard/CartCard.js]: done

File: src/App.js (hunk 1 above)
Comment (krgarima) [reply src/App.js]: done

**Pull request merged**